modules/gateway/peers.go

```diff
@@ -175,6 +175,9 @@ func (g *Gateway) Connect(addr modules.NetAddress) error {
 	if addr == g.Address() {
 		return errors.New("can't connect to our own address")
 	}
+	if build.Release != "testing" && addr.IsLoopback() {
```

Comment:
On a related note, I'd like to eliminate as many `build.Release != "testing"` statements as possible. Mocking should make this more feasible.